test/unit/test.sign.js

```diff
@@ -648,17 +649,69 @@ describe('sign', function() {
     }, options);
 
     var program = {};
-
-    return signCmd(manifest, program, options.cmdOptions, {
+    var cmdConfig = {
       systemProcess: mockProcess,
       AMOClient: options.StubAMOClient,
-    });
+    };
+    if (options.getManifest !== null) {
+      cmdConfig.getManifest = options.getManifest;
+    }
+
+    return signCmd(program, options.cmdOptions, cmdConfig);
   }
 
   it('should exit 0 on signing success', function(done) {
     runSignCmd().then(function() {
-      expect(wasSigned).to.be.equal(true);
+      expect(signingCall.wasCalled).to.be.equal(true);
+      expect(mockProcessExit.call[0]).to.be.equal(0);
+      done();
+    }).catch(done);
+  });
+
+  it('passes manifest info to the signer', function(done) {
+    runSignCmd().then(function() {
+      expect(signingCall.wasCalled).to.be.equal(true);
+      // This checks that version/guid values from
+      // the manifest (loaded from a fixture) are used.
+      expect(signingCall.call[0].version).to.be.equal('1.0.0');
+      expect(signingCall.call[0].guid).to.be.equal('@simple-addon');
+      done();
+    }).catch(done);
+  });
+
+  it('throws an error for XPI file errors', function(done) {
+    runSignCmd({
+      cmdOptions: {
+        xpi: "/not/a/real/path.xpi",
+        apiKey: 'some-key',
+        apiSecret: 'some-secret',
+      },
+    }).then(function() {
+      expect(mockProcessExit.call[0]).to.be.equal(1);
+      done();
+    }).catch(done);
+  });
+
+  it('passes custom XPI to the signer', function(done) {
+    var mockManifestGetter = new CallableMock({
+      returnValue: when.promise(function(resolve) {
+        resolve({});  // resolve with an empty manifest.
+      }),
+    });
+    // Make sure nothing is checking the working directory for add-on things.
+    process.chdir(path.join(__dirname, "..", "addons"));
```

Comment:
I don't see it switching back. Is this safe? Can you put a real XPI on the filesystem and skip the mock?